src/data/users.go

```diff
@@ -0,0 +1,68 @@
+package data
+
+import (
+	"errors"
+	"fmt"
+	"github.com/go-ldap/ldap"
+	"github.com/portasonic/backend/src/directory"
+)
+
+var (
+	ErrLoginFailed = errors.New("login failed")
+)
+
+type UserRepository struct {
+	ctx *persistenceContext
+}
+
+func (u UserRepository) Login(username string, password string) (string, error) {
+	ldapCon, err := directory.Open(u.ctx.config.Ldap)
+	if err != nil {
+		return "", fmt.Errorf("could not connect to user directory: %w", err)
+	}
+
+	defer func() {
+		ldapCon.Close()
+	}()
```

Comment:
it's just a detail, but why is that in an anonymous function? 

That would be simpler:  
```suggestion
	defer ldapCon.Close()
```